mne/tests/test_source_estimate.py

```diff
@@ -2054,3 +2054,31 @@ def test_label_extraction_subject(kind):
         stc.subject = None
         with pytest.raises(ValueError, match=r"label\.sub.*not match.* sour"):
             extract_label_time_course(stc, labels_fs, src)
+
+
+def test_apply_function_stc():
+    """Check the apply_function method for source estimate data."""
+    # Create a sample _BaseSourceEstimate object
+    n_vertices = 100
+    n_times = 200
+    vertices = [np.array(np.arange(50)), np.array(np.arange(50, 100))]
+    tmin = 0.0
+    tstep = 0.001
+    data = np.random.rand(n_vertices, n_times)
+
+    stc = _make_stc(data, vertices, tmin=tmin, tstep=tstep, src_type="surface")
+
+    # A sample function to apply to the data
+    def fun(data_row, **kwargs):
+        return 2 * data_row
+
+    # Test applying the function to all vertices without parallelization
+    stc_copy = stc.copy()
+    stc.apply_function(fun)
+    for idx in range(n_vertices):
+        assert_array_equal(stc._data[idx, :], 2 * stc_copy._data[idx, :])
```

Comment:
Always prefer public attributes when possible, and for float arithmetic it's a good idea to use `assert_allclose` rather than `assert_array_equal`
```suggestion
        assert_allclose(stc.data[idx, :], 2 * stc_copy.data[idx, :])
```